server/models/user.js

```diff
@@ -53,30 +53,28 @@ UserSchema.pre('save', function (next) {
   }
 });
 
-UserSchema.statics.authenticate = function (email, password) {
-  return User.findOne({ email })
+
+UserSchema.statics.authenticate = function authenticate(email, password) {
+  return this.findOne({ email })
     .exec()
-    .then(user => {
+    .then((user) => {
       if (user && bcrypt.compareSync(password, user.password)) {
         return user;
-      } else {
-        throw {
-          error: {
-            message: 'Incorrect email and password combination.',
-            name: 'AuthenticationError'
-          }
-        };
       }
+      throw new Error(JSON.stringify({
+        error: {
+          message: 'Incorrect email and password combination.',
+          name: 'AuthenticationError',
+        },
+      }));
```

Comment:
ESLint requires `Error()` object to be thrown. This is my current solution, but I wonder if it would make more sense to skip the `JSON.stringify()` and create something like:
 `const e = new Error('Incorrect email and password combination.')`
 `e.name = 'AuthenticationError'`
 `throw e`
Then, adjust the front-end to pickup the Error object in the response. Does that make more sense?